diff --git a/backends/imgui_impl_dx12.cpp b/backends/imgui_impl_dx12.cpp
index 1f3cc8990..b2c21b71a 100644
--- a/backends/imgui_impl_dx12.cpp
+++ b/backends/imgui_impl_dx12.cpp
@@ -479,6 +479,17 @@ bool    ImGui_ImplDX12_CreateDeviceObjects()
                 return false;
         }
 
+#ifdef __MINGW32__
+        PFN_D3D12_SERIALIZE_VERSIONED_ROOT_SIGNATURE D3D12SerializeVersionedRootSignatureFn =
+            (PFN_D3D12_SERIALIZE_VERSIONED_ROOT_SIGNATURE)::GetProcAddress(d3d12_dll, "D3D12SerializeVersionedRootSignature");
+        if (D3D12SerializeVersionedRootSignatureFn == NULL)
+            return false;
+
+        ID3DBlob* blob = NULL;
+        D3D12_VERSIONED_ROOT_SIGNATURE_DESC versioned_desc = {.Version = D3D_ROOT_SIGNATURE_VERSION_1_0, .Desc_1_0 = desc};
+        if (D3D12SerializeVersionedRootSignatureFn(&versioned_desc, &blob, NULL) != S_OK)
+            return false;
+#else
         PFN_D3D12_SERIALIZE_ROOT_SIGNATURE D3D12SerializeRootSignatureFn = (PFN_D3D12_SERIALIZE_ROOT_SIGNATURE)::GetProcAddress(d3d12_dll, "D3D12SerializeRootSignature");
         if (D3D12SerializeRootSignatureFn == NULL)
             return false;
@@ -486,6 +497,7 @@ bool    ImGui_ImplDX12_CreateDeviceObjects()
         ID3DBlob* blob = NULL;
         if (D3D12SerializeRootSignatureFn(&desc, D3D_ROOT_SIGNATURE_VERSION_1, &blob, NULL) != S_OK)
             return false;
+#endif
 
         g_pd3dDevice->CreateRootSignature(0, blob->GetBufferPointer(), blob->GetBufferSize(), IID_PPV_ARGS(&g_pRootSignature));
         blob->Release();
diff --git a/backends/imgui_impl_dx12.h b/backends/imgui_impl_dx12.h
index 7051962a0..1c3ecbf7e 100644
--- a/backends/imgui_impl_dx12.h
+++ b/backends/imgui_impl_dx12.h
@@ -21,6 +21,8 @@
 #pragma warning (disable: 4471) // a forward declaration of an unscoped enumeration must have an underlying type
 #endif
 
+#include <dxgiformat.h>
+
 enum DXGI_FORMAT;
 struct ID3D12Device;
 struct ID3D12DescriptorHeap;
